pkg/controller/atlasproject/team_reconciler.go

```diff
@@ -149,44 +150,25 @@ func (r *AtlasProjectReconciler) teamReconcile(
 	}
 }
 
-func createTeamContextFromParent(
-	ctx context.Context,
-	team *v1.AtlasTeam,
-	kubeClient client.Client,
-	atlasConnection atlas.Connection,
-	atlasDomain string,
-	logger *zap.SugaredLogger,
-) (*workflow.Context, error) {
-	teamCtx := customresource.MarkReconciliationStarted(kubeClient, team, logger, ctx)
-	teamCtx.Connection = atlasConnection
-	atlasClient, err := atlas.Client(atlasDomain, atlasConnection, logger)
-	if err != nil {
-		return nil, err
-	}
-	teamCtx.Client = atlasClient
-
-	return teamCtx, nil
-}
-
-func ensureTeamState(workflowCtx *workflow.Context, team *v1.AtlasTeam) (string, workflow.Result) {
+func ensureTeamState(workflowCtx *workflow.Context, orgID string, team *v1.AtlasTeam) (string, workflow.Result) {
 	var atlasTeam *mongodbatlas.Team
 	var err error
 
 	if team.Status.ID != "" {
-		atlasTeam, err = fetchTeamByID(workflowCtx, team.Status.ID)
+		atlasTeam, err = fetchTeamByID(workflowCtx, orgID, team.Status.ID)
```

Comment:
I think orgID was better within the custom context object, that context is for a reconciliation and it will NOT have a different value. I Understand the refactor affects it, but I would have prefered a way to leave it in the context somehow.

Reply:
I have added orgID to the workflow context that is passed around the code.